File: src/main/java/com/micahelias/opengl/VertexBuffer.java
package com.micahelias.opengl;

import static org.lwjgl.opengl.GL15.*;

/**
 * VertexBuffer
 */
public class VertexBuffer {

  private int m_RendererID;
  
  public VertexBuffer(float[] data) {
    m_RendererID = glGenBuffers();
    glBindBuffer(GL_ARRAY_BUFFER, m_RendererID);
    glBufferData(GL_ARRAY_BUFFER, data, GL_STATIC_DRAW);
  }


  void bind() {
    glBindBuffer(GL_ARRAY_BUFFER, m_RendererID);
  }

  void unbind() {
    glBindBuffer(GL_ARRAY_BUFFER, 0);
  }

}

File: mango/opengl/vertex_buffer.go
package opengl

import (
	"github.com/go-gl/gl/v3.3-core/gl"
)

type VertexBuffer struct {
	m_RendererID uint32
}

func NewVertexBuffer(data []float32) *VertexBuffer {
	var vb uint32
	gl.GenBuffers(1, &vb)
	gl.BindBuffer(gl.ARRAY_BUFFER, vb)
	gl.BufferData(gl.ARRAY_BUFFER, 4*len(data), gl.Ptr(&(data[0])), gl.STATIC_DRAW)
	return &VertexBuffer{m_RendererID: vb}
}

func (vb *VertexBuffer) Bind() {
	gl.BindBuffer(gl.ARRAY_BUFFER, vb.m_RendererID)
}

func (vb *VertexBuffer) Unbind() {
	gl.BindBuffer(gl.ARRAY_BUFFER, 0)
}
